File: service/Api/Question/Question.cs
using System;
using System.Collections.Generic;

namespace Quizmeister
{
    public class Question
    {
        public int QuestionId { get; set; }
        public string ShortId { get; set; }
        public List<Answer> Answers { get; set; }
        public string Title { get; set; }
        public QuestionTypes Type { get; set; }
        public string Remark { get; set; }
        public DateTime ModifiedOn { get; set; }
        public Scoring Scoring { get; set; }
        public string SearchField { get; set; }
        public List<Round> Rounds { get; set; }
    }
}

File: service/src/question/question.js
const mongoose = require('mongoose');

const answerSchema = new mongoose.Schema({
  text: String,
  correct: Boolean,
  searchField: String,
});
answerSchema.virtual('answerId').get(function() { return this._id.toHexString(); });
answerSchema.set('toJSON', {
  virtuals: true,
  versionKey:false,
  transform: function (doc, ret) {   delete ret._id;delete ret.id;  },
});

const scoringSchema = new mongoose.Schema({
  type: Number,
  weightCorrectAnswer: Number,
  weightNoAnswer: Number,
  timeLimitSeconds: Number,
  timeScoringInterval: Number,
});
scoringSchema.virtual('scoringId').get(function() { return this._id.toHexString(); });
scoringSchema.set('toJSON', {
  virtuals: true,
  versionKey:false,
  transform: function (doc, ret) {   delete ret._id;delete ret.id;  },
});

const questionSchema = new mongoose.Schema({
  shortId: String,
  title: String,
  remark: String,
  type: Number, 
  answers: [answerSchema],
  scoring: scoringSchema,
  modifiedOn: Date,
  searchField: String,
});
questionSchema.virtual('questionId').get(function() { return this._id.toHexString(); });
questionSchema.set('toJSON', {
  virtuals: true,
  versionKey:false,
  transform: function (doc, ret) {   delete ret._id;delete ret.id;  },
});

const Question = mongoose.model('Question', questionSchema);

module.exports = {
  Question
}